new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainOnPV.java

```diff
@@ -0,0 +1,700 @@
+                        new V1LocalObjectReference()
+                            .name("docker-store"))))));
+    String jobName = assertDoesNotThrow(() -> TestActions
+        .createNamespacedJob(jobBody), "Domain creation job failed");
+
+    logger.info("Checking if the domain creation job {0} completed in namespace {1}",
+        jobName, namespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for job {0} to be completed in namespace {1} "
+                + "(elapsed time {2} ms, remaining time {3} ms)",
+                jobName,
+                namespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(jobCompleted(jobName, null, namespace));
+  }
+
+  /**
+   * Create secret for docker credentials.
+   */
+  private void createDockerSecret() {
+    // docker login, if necessary
+    if (!REPO_USERNAME.equals(REPO_DUMMY_VALUE)) {
+      logger.info("docker login");
+      assertTrue(dockerLogin(REPO_REGISTRY, REPO_USERNAME, REPO_PASSWORD), "docker login failed");
+    }
+
+    // Create the V1Secret configuration
+    V1Secret repoSecret = new V1Secret()
+        .metadata(new V1ObjectMeta()
+            .name(REPO_SECRET_NAME)
+            .namespace(domainNamespace))
+        .type("kubernetes.io/dockerconfigjson")
+        .putDataItem(".dockerconfigjson", dockerConfigJson.getBytes());
+
+    boolean secretCreated = assertDoesNotThrow(() -> createSecret(repoSecret),
+        String.format("createSecret failed for %s", REPO_SECRET_NAME));
+    assertTrue(secretCreated, String.format("createSecret failed while creating secret %s", REPO_SECRET_NAME));
+  }
+
+  /**
+   * Create secret for WebLogic credentials.
+   */
+  private void createWebLogicCredentialsSecret() {
+    logger.info("Creating secret for WebLogic credentials");
+    wlSecretName = "weblogic-credentials";
+    Map<String, String> adminSecretMap = new HashMap<>();
+    adminSecretMap.put("username", adminUser);
+    adminSecretMap.put("password", adminPassword);
+    boolean secretCreated = assertDoesNotThrow(() -> createSecret(new V1Secret()
+        .metadata(new V1ObjectMeta()
+            .name(wlSecretName)
+            .namespace(domainNamespace))
+        .stringData(adminSecretMap)), "Create secret failed with ApiException");
+    assertTrue(secretCreated, String.format("create secret failed for %s", wlSecretName));
+  }
+
+  /**
+   * Create a persistent volume and persistent volume claim.
+   * @throws IOException when creating pv path fails
+   */
+  private void createPVandPVC() throws IOException {
+    logger.info("creating persistent volume and persistent volume claim");
+
+    Path pvHostPath = Files.createDirectories(Paths.get(
+        TestConstants.PV_ROOT, this.getClass().getSimpleName(), domainUid + "-persistentVolume"));
+    logger.info("Creating PV directory {0}", pvHostPath);
```

Comment:
This is good.  Just putting this comment here so that you know that for some kinds of clusters, such as OKE & OpenShift, the tests will not have direct file system access to the host path. We will need to come and change to later to run tests against OKE or OpenShift. In those cases, we will likely have an NFS mount and you won't be able to access the files from this code without doing something like running another job that mounts the pv/pvc.

Reply:
Got it.